packages/react/src/components/SuiteHeader/IdleLogoutConfirmationModal/IdleLogoutConfirmationModal.test.jsx

```diff
@@ -0,0 +1,142 @@
+import { render, screen, within } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import '@testing-library/jest-dom/extend-expect';
+import React from 'react';
+
+import SuiteHeaderI18N from '../i18n';
+
+import IdleLogoutConfirmationModal from './IdleLogoutConfirmationModal';
+
+const commonProps = {
+  routes: {
+    logout: 'https://www.ibm.com',
+    logoutInactivity: 'https://www.ibm.com',
+    domain: '',
+  },
+  idleTimeoutData: { countdown: 10, timeout: 10, cookieName: '_user_inactivity_timeout' },
+};
+
+describe('IdleLogoutConfirmationModal', () => {
+  let originalWindowLocation;
+  beforeEach(() => {
+    jest.useFakeTimers();
+    originalWindowLocation = { ...window.location };
+    delete window.location;
+    window.location = { href: '' };
+    window.open = jest.fn();
+  });
+
+  afterEach(() => {
+    window.location = { ...originalWindowLocation };
+    jest.useRealTimers();
+  });
+
+  it('active user does not see idle logout confirmation dialog', () => {
+    render(<IdleLogoutConfirmationModal {...commonProps} />);
+    // Simulate a timestamp cookie that is in the future
+    Object.defineProperty(window.document, 'cookie', {
+      writable: true,
+      value: `${commonProps.idleTimeoutData.cookieName}=${Date.now() + 1000}`,
+    });
+    jest.runOnlyPendingTimers();
+    expect(screen.getByTestId('idle-logout-confirmation')).not.toHaveClass('is-visible');
+  });
+  it('inactive user sees idle logout confirmation dialog', () => {
+    render(<IdleLogoutConfirmationModal {...commonProps} />);
+    // Simulate a timestamp cookie that is in the past
+    Object.defineProperty(window.document, 'cookie', {
+      writable: true,
+      value: `${commonProps.idleTimeoutData.cookieName}=${Date.now() - 1000}`,
+    });
+    jest.runOnlyPendingTimers();
+    expect(screen.getByTestId('idle-logout-confirmation')).toHaveClass('is-visible');
+  });
+  it('inactive user does not see idle logout confirmation dialog if timeout is set to 0', () => {
+    render(
+      <IdleLogoutConfirmationModal
+        {...commonProps}
+        idleTimeoutData={{ ...commonProps.idleTimeoutData, timeout: 0 }}
+      />
+    );
+    // Simulate a timestamp cookie that is in the past
+    Object.defineProperty(window.document, 'cookie', {
+      writable: true,
+      value: `${commonProps.idleTimeoutData.cookieName}=${Date.now() - 1000}`,
+    });
```

Comment:
same cookie reset in after each.

Reply:
Done.
And I've also made the same change in `IdleTimer.test.js`